server/service/osquery_utils/queries.go

```diff
@@ -754,6 +760,46 @@ func directIngestBattery(ctx context.Context, logger log.Logger, host *fleet.Hos
 	return ds.ReplaceHostBatteries(ctx, host.ID, mapping)
 }
 
+func directIngestWindowsUpdateHistory(
+	ctx context.Context,
+	logger log.Logger,
+	host *fleet.Host,
+	ds fleet.Datastore,
+	rows []map[string]string,
+	failed bool,
+) error {
+	if failed {
+		level.Error(logger).Log("op", "directIngestWindowsUpdateHistory", "err", "failed")
+		return nil
+	}
+
+	// The windows update history table will also contain entries for the Defender Antivirus. Unfortunately
+	// there's no reliable way to differentiate between those entries and Cumulative OS updates.
+	// Since each antivirus update will have the same KB ID, but different 'dates', to
+	// avoid trying to insert duplicated data, we group by KB ID and then take the most 'out of
+	// date' update in each group.
```

Comment:
Curious as to why use the most 'out of date' update?

Reply:
To avoid issuing un-necessary `insert` statements when dealing with Antivirus updates. If we were to take the most 'up to date' entry then the logic [here](https://github.com/fleetdm/fleet/pull/7138/files#diff-b4b9c78a4fca9ec372673f078286ddd316f46ce6510850bc93b11dbb69097a8fR36) will generate insert statements for each new Antivirus update - which we don't want.